aws/data_source_aws_ami_ids_test.go

```diff
@@ -7,7 +7,7 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 )
 
-func TestAccDataSourceAwsAmiIds_basic(t *testing.T) {
+func TestAccEC2AMIIDsDataSource_basic(t *testing.T) {
```

Comment:
What determines whether test naming should include a fully capitalized initialism? Should we enforce the standard?

Reply:
Unlike above, the [Go convention](https://github.com/golang/go/wiki/CodeReviewComments#initialisms) is clear on this point:

> Words in names that are initialisms or acronyms (e.g. "URL" or "NATO") have a consistent case. For example, "URL" should appear as "URL" or "url" (as in "urlPony", or "URLPony"), never as "Url".

Although the AWS SDK for Go breaks this rule a lot, I have seen efforts to change course and some of the newer APIs follow the rules. My vote is that we stick with convention and win the respect of Go coders.

Reply:
If we are going to enforce this, we will need to manually enumerate all initialisms and maintain the list going forward. For example, what does `sesv2` convert into?

I would prefer to see a proposal on that before really going down this route since it affects all code and testing contributions, otherwise it will be easy to escape out of the desired convention. I'm guessing it would need to be a text file with a list or a Go file with a slice of strings, but we will need to figure out how to ensure it is usable by tooling that needs it.

For what it is worth, not using initialisms allows you to easily convert snake_case to PascalCase and lowercase to Titlecase without additional logic to also bake in the conversions. Not saying that needs to be an end-all goal, but I'm not sure if folks are really chasing after this project's code for "correctness" in this sense. More likely folks are probably interested in functionality that is readable and works.

Reply:
I'm specifically limiting the scope of changes to test names. We have non-breaking freedom with test names. However, fixing things in test names but not code introduces a level of inconsistency. For instance, removing "AWS" and "Aws" from test names but not all the resource names. Despite this issue, which I think is small, I love the idea of having one area where we have clean, Go convention adhereing names.

Following Go's conventions, the "V" would either be the next word and capitalized or an initial and capitalized:

package | testing name
-- | --
apigatewayv2 | APIGatewayV2
cloudhsmv2 | CloudHSMV2
elbv2 | ELBV2
greengrassv2 | GreengrassV2
kinesisanalyticsv2 | KinesisAnalyticsV2
lexmodelsv2 | LexModelsV2
lexruntimev2 | LexRuntimeV2
sesv2 | SESV2
wafv2 | WAFV2

Reply:
I like the `<service>V2` approach in general. However, it doesn't look right after initialisms, such as `WAFV2`. For straight up consistency, I'd go with `<service>V2`. For ease of readability or aesthetics, I'd go with `<full word service>V2` and `<initialism service>v2`.

Reply:
We can extract the list of initialisms from the SDK packages at `service/<service package>/service.go` files. The constant `ServiceID` could be checked for all-caps. There are some special cases, such as `ACM PCA`, where there's a space.

Reply:
This also somewhat relates to another cleanup issue I'd like to open around standardizing service and resource naming in, e.g. error messages